File: parties-and-accounts/get-parties.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega node (REST)

Apps/Libraries:
- REST: requests (https://pypi.org/project/requests/)

Responses:
- response-examples.txt
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import json
import os
import requests
import helpers

node_url_rest = os.getenv("NODE_URL_REST")
if not helpers.check_url(node_url_rest):
    print("Error: Invalid or missing NODE_URL_REST environment variable.")
    exit(1)

# __get_parties:
# Request a list of parties trading on a Vega network
url = "{base}/parties".format(base=node_url_rest)
response = requests.get(url)
helpers.check_response(response)
response_json = response.json()
print("Parties:\n{}".format(json.dumps(response_json, indent=2, sort_keys=True)))
# :get_parties__

assert len(response_json["parties"]) > 0
pubkey = response_json["parties"][0]["id"]

# __get_party_by_id:
# Request a party by their identifier (this is their public key)
url = "{base}/parties/{partyID}".format(base=node_url_rest, partyID=pubkey)
response = requests.get(url)
helpers.check_response(response)
response_json = response.json()
print("PartyById:\n{}".format(json.dumps(response_json, indent=2, sort_keys=True)))
# :get_party_by_id__

File: parties-and-accounts/get-parties.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/vegaprotocol/api-clients/go/generated/code.vegaprotocol.io/vega/proto/api"
	"google.golang.org/grpc"
)

func main() {
	nodeURLGrpc := os.Getenv("NODE_URL_GRPC")
	if len(nodeURLGrpc) == 0 {
		panic("NODE_URL_GRPC is null or empty")
	}

	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	dataClient := api.NewTradingDataServiceClient(conn)
	// __get_parties:
	// Request a list of parties trading on a Vega network
	request := api.PartiesRequest{}
	parties, err := dataClient.Parties(context.Background(), &request)
	if err != nil {
		panic(err)
	}

	fmt.Printf("Parties: %v\n", parties)
	// :get_parties__

	partyID := parties.Parties[0].Id
	// __get_party_by_id:
	// Request a party by their identifier (this is their public key)
	partyByIDReq := api.PartyByIDRequest{PartyId: partyID}
	partyByIDResp, err := dataClient.PartyByID(context.Background(), &partyByIDReq)
	if err != nil {
		panic(err)
	}

	fmt.Printf("PartyById: %v\n", partyByIDResp.Party)
	// :get_party_by_id__
}
